testing/test_faulthandler.py

```diff
@@ -110,6 +110,39 @@ def test_timeout():
     assert result.ret == 0
 
 
+@pytest.mark.skip(reason="sometimes crashes on CI (#7022)")
+@pytest.mark.parametrize("exit_on_timeout", [True, False])
+def test_timeout_and_exit(pytester: Pytester, exit_on_timeout: bool) -> None:
+    """Test option to force exit pytest process after a certain timeout."""
+    pytester.makepyfile(
+        """
+    import os, time
+    def test_long_sleep_and_raise():
+        time.sleep(1 if "CI" in os.environ else 0.1)
```

Comment:
Shall I remove the `if "CI" in os.environ` condition that can never hold as long as pytest's CI is configured to not propagate this env var via its tox.ini config?

Reply:
When I tried to add `CI` to `passenv` (see 000876e above in this PR) this broke many other unrelated tests:

https://github.com/pytest-dev/pytest/actions/runs/17271882336/job/49018562088

Reply:
I wonder if the `is_ci_env` could be computed early, before the test runs instead of having this check within. It could be checked before the tests implement shielding through `os.environ` manipulation. Perhaps, a session fixture in `conftest.py` would be useful in general. Thoughts?

Reply:
For this particular case, once the CI variable is properly propagated, just increasing the sleep time as originally intended might be enough to make the test stable again and we would not have skip those tests anymore.